transports/tls/Cargo.toml

```
@@ -0,0 +1,29 @@
+[package]
+name = "libp2p-tls"
+version = "0.1.0"
```

```suggestion
version = "0.1.0-alpha"
```

To the best of my knowledge, the code below has not run in any production systems.

To communicate expectations, I suggest releasing this as an alpha first.

See also same suggestion on QUIC pull request https://github.com/libp2p/rust-libp2p/pull/2289#discussion_r983395958

---

On it!

Most users depend on `libp2p` though so they are not going to see this version. IMO we are effectively removing that "alpha" label again through the re-export.

Do you think we should perhaps not re-export the TLS implementation yet so users have to depend on it manually?

---

> Most users depend on `libp2p` though so they are not going to see this version. IMO we are effectively removing that "alpha" label again through the re-export.

That is true. Though I would explicitly call it out in the changelog.

> Do you think we should perhaps not re-export the TLS implementation yet so users have to depend on it manually?

I don't have an opinion here. I think we should be consistent with `libp2p-quic`. Chatted with @elenaf9 quickly. She prefers not removing `libp2p-quic` from `libp2p`.

---

What do you think about annotating the re-export with the [`unstable` attribute macro](https://rustc-dev-guide.rust-lang.org/stability.html#unstable) @thomaseizinger ?

---

> What do you think about annotating the re-export with the [`unstable` attribute macro](https://rustc-dev-guide.rust-lang.org/stability.html#unstable) @thomaseizinger ?

That is specific to the Rust compiler from what I know, I don't think you can use that actually?

---

Ah yes you are right! Don't have a strong on opinion on whether to re-export tls and quic or not. I did not consider that when re-exporting we hide the alpha-flag, so maybe not re-exporting is the best solution after all.

---

I think not exporting it would be the better solution but I don't want to block this PR on it. We can always change that until the next release.